### DISCUSSION TITLE
MQTT v5.0

### DISCUSSION THREAD

**Discussion by MatteoPiovanelli** [strimzi/strimzi-kafka-operator#9543] [Q&A]
Hello,

We are looking into this project and were wondering how it supports MQTT v5.0.
Messages on our queue use that version of the protocol, and make use of some custom properties.

The MqttDecoder appears to support it, as long as it is specified on connection.
The readme on your main repo page only mentions MQTT v3.1.1.

Is there any specific thing I should take care of doing to make sure I don't "lose" any data, and the sent properties are all received into Kafka?

Sorry if this may be a basic question, but this is my first "real" MQTT project. (and right now I don't have the infrastructure available to set things up and just test things myself).

Thank you.

**Comment by llukyanov** [comment]
Hi @MatteoPiovanelli-Laser  I don't believe Strimzi bridge supports MQTT v5. If you specifically need v5, the Zilla proxy does support it: https://github.com/aklivity/zilla?tab=readme-ov-file#-mqtt-kafka-proxying (full disclosure I'm one of the maintainers of the Zilla project)
